### PR TITLE
`in` operator should only return false if key is actually missing.

### PR DESCRIPTION
**Pull request by swist**
Addresses #80. I'm not sure if I like the solution with a sentinel, but seems to be in line with your philosophy of doing minimum possible changes

Do you see a better way other than special_casing the `select()` behaviour? It seems like is used in more than one way

### CODE REVIEW

File: tests/test_basic_ops_dict.py [hunk 1]
```diff
@@ -142,9 +142,9 @@ def test_dict_keys():
 def test_pickle_get_root():
     # Test that get_root() is reconstructed correctly for pickle loaded files.
     with tempfile.TemporaryFile() as fp:
-        c1 = OmegaConf.create(dict(a=dict(a1=1, a2=2,),))
+        c1 = OmegaConf.create(dict(a=dict(a1=1, a2=2)))
 
-        c2 = OmegaConf.create(dict(b=dict(b1="???", b2=4, bb=dict(bb1=3, bb2=4,),),))
+        c2 = OmegaConf.create(dict(b=dict(b1="???", b2=4, bb=dict(bb1=3, bb2=4))))
```
Comment (swist) [inline tests/test_basic_ops_dict.py]: I don't know why these got caught in `black` crossfire

**Comment by omry** [comment]
Thanks!
I have no such philosophy, I consider the unit tests the guardians against bad changes. 

I just spotted this bug yesterday during a hackathon so I didn't think much about the solution.
I did think it might be tricky, and I thought that the solution might involve splitting select into two functions, maybe:
```
def _dereference_key(key):
   # return key if not an interpolation
   # else return the key pointed to by key, in the test case this would return "a" -> "a" and "b" -> "a"
   # also "x.b" -> "x.a" in a test like {"x": {"a": 10, "b": "${x.a}"}}
```

select can then be implemented as something like:
```
def select(self, key):
  real_key = self._dereference_key(key)
  # current select logic on real_key
```

Then contains can be implemented as:
```
def __contains__(self , key )
   real_key = self._dereference_key(key)
   # contains on real_key
```

Thoughts?

**Comment by omry** [comment]
@swist, I created a chat for Hydra and OmegaConf (see main readme). would be great if you can join.

**Comment by swist** [comment]
I have tried something similar - adding `exists` which behaves similarly to `select` but doesn't return the value, but wasn't happy with that. Perhaps delegating a bit of work that select currently does will work better

**Comment by omry** [comment]
Yeah, this is what I had in mind.
Need to see what the code would actually look like when implemented to really evaluate if this is cleaner than you proposed solution.

**Comment by omry** [comment]
One thing that guides me in general is to be careful exposing new public APIs (those are easy to introduce and hard to remove).
Any reason you went with this and not with my proposed private API?

exists is not a good name here, it sounds like __contains__(self, key).
I think _dereference_key is better here, but as I said it may be tricky to implement cleanly.
Since select can be implemented in terms of _dereference_key() this will also minimize code duplication.

**Comment by swist** [comment]
Didn't really mean to make it a public API, it was just me digging through the reflog so I could show you what a half-way shop to `_dereference_key` would look like
If you think that this is a better direction than sentinels I'll have a look at `_dereference_key` sometime this week

**Comment by omry** [comment]
Oh, gotcha.
Yeah, my gut feeling is that this should result in a cleaner change, despite the clunky name for _dereference_key().
Thanks again.

**Comment by omry** [comment]
FYI:
I plan to release a new version of OmegaConf either this weekend or next week, this is the last bug I want to address before this new version.

Do you think you will be able to finish this PR soon?
If you don't have time I don't mind taking over.

**Comment by swist** [comment]
I was going to look at it today!

File: tests/test_basic_ops_dict.py (hunk 1 above)
Comment (omry) [reply tests/test_basic_ops_dict.py]: Interesting, black does not fix this for me.
On the other hand, it's also fine with the version without the extra commas.

**Comment by swist** [comment]
fixed in #84

**Pull request closed without merging**